src/bun.js/node/node_fs_watcher.zig

```diff
@@ -540,241 +562,202 @@ pub const FSWatcher = struct {
 
         pub fn createFSWatcher(this: Arguments) !JSC.JSValue {
             const obj = try FSWatcher.init(this);
-            return obj.toJS();
+            if (obj.js_this != .zero) {
+                return obj.js_this;
+            }
+            return JSC.JSValue.jsUndefined();
         }
     };
 
-    pub const JSObject = struct {
-        signal: ?*JSC.AbortSignal,
-        persistent: bool,
-        manager: ?*FSWatcher.Watcher,
-        fsevents_watcher: ?*FSEvents.FSEventsWatcher,
-        poll_ref: JSC.PollRef = .{},
-        globalThis: ?*JSC.JSGlobalObject,
-        js_this: JSC.JSValue,
-        encoding: JSC.Node.Encoding,
-        closed: bool,
-
-        pub usingnamespace JSC.Codegen.JSFSWatcher;
-
-        pub fn getFSWatcher(this: *JSObject) *FSWatcher {
-            if (this.manager) |manager| return manager.ctx;
-            if (this.fsevents_watcher) |manager| return bun.cast(*FSWatcher, manager.ctx.?);
-
-            @panic("No context attached to JSFSWatcher");
+    pub fn initJS(this: *FSWatcher, listener: JSC.JSValue) void {
+        if (this.persistent) {
+            this.poll_ref.ref(this.ctx);
         }
 
-        pub fn init(globalThis: *JSC.JSGlobalObject, manager: ?*FSWatcher.Watcher, fsevents_watcher: ?*FSEvents.FSEventsWatcher, signal: ?*JSC.AbortSignal, listener: JSC.JSValue, persistent: bool, encoding: JSC.Node.Encoding) !*JSObject {
-            var obj = try globalThis.allocator().create(JSObject);
-            obj.* = .{
-                .signal = null,
-                .persistent = persistent,
-                .manager = manager,
-                .fsevents_watcher = fsevents_watcher,
-                .globalThis = globalThis,
-                .js_this = .zero,
-                .encoding = encoding,
-                .closed = false,
-            };
-            const instance = obj.getFSWatcher();
-
-            if (persistent) {
-                obj.poll_ref.ref(instance.ctx);
-            }
+        this.js_this = FSWatcher.toJS(this, this.globalThis);
```

Comment:
This will need to be assigned to the stack so that the GC can see it.


```suggestion
        const js_this = FSWatcher.toJS(this, this.globalThis);
        js_this.ensureStillAlive();
        this.js_this = js_this;
```

Reply:
fixed into https://github.com/oven-sh/bun/pull/3396/commits/feb30ca7197f17b39c0057dc6a0c455b2964ff89